src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/SafeTransactionPreview.tsx

```diff
@@ -119,84 +138,173 @@ interface Props {
         </div>
       </DialogSection>
-      <DialogSection>
-        <div className={styles.transactionDetailsSection}>
-          {values.transactions[0].transactionType !== '' && (
-            <div className={styles.transactionTitle}>
-              1.{' '}
-              <FormattedMessage
-                {...ConstantsMSG[values.transactions[0].transactionType]}
+
+      {values.transactions.map((transaction, index) => (
+        <div key={autogeneratedIds[index]}>
+          {values.transactions.length > 1 && (
+            <div
+              className={classnames(styles.transactionHeading, {
+                [styles.transactionHeadingOpen]: transactionTabStatus[index],
+              })}
+            >
+              <Heading
+                appearance={{
+                  size: 'normal',
+                  margin: 'none',
+                  theme: 'dark',
+                }}
+                text={MSG.transactionTitle}
+                textValues={{
+                  transactionNumber: index + 1,
+                  transactionType: getTransactionTypeLabel(
+                    transaction.transactionType,
+                    /* Need casting as `formatValues` func used in Heading
+                            doesn't have correct types (coming from react-intl) */
+                  ) as PrimitiveType,
+                }}
               />
+              <Button
+                className={styles.tabButton}
+                onClick={() => handleTabToggle(index)}
+              >
+                <Icon
+                  name="caret-up-small"
+                  className={styles.toggleTabIcon}
+                  title={MSG.toggleTransaction}
+                  titleValues={{
+                    tabToggleStatus: transactionTabStatus[index],
+                  }}
+                />
+              </Button>
             </div>
           )}
-          <DetailsItem
-            label={MSG.safe}
-            value={
-              <AddressDetailsView
-                item={(values.safe as never) as AnyUser}
-                isSafeItem
-              />
-            }
-          />
-          {values.transactions[0].transactionType !==
-            TransactionTypes.CONTRACT_INTERACTION && (
-            <DetailsItem
-              label={MSG.to}
-              value={
-                <AddressDetailsView
-                  item={(values.transactions[0].recipient as never) as AnyUser}
-                  isSafeItem={false}
+          <div
+            className={classnames({
+              [styles.tabContentClosed]:
+                values.transactions.length > 1 && !transactionTabStatus[index],
+            })}
+          >
+            <DialogSection>
+              <div className={styles.transactionDetailsSection}>
+                {values.transactions[index].transactionType !== '' &&
+                  values.transactions.length === 1 && (
+                    <div className={styles.transactionTitle}>
+                      1.{' '}
```

Comment:
I see that @arren has already answered a question about style & format, so this will be the standard numbering approach.